core/src/main/java/org/geysermc/geyser/entity/type/player/SessionPlayerEntity.java

```diff
@@ -109,10 +110,17 @@ public void setPositionManual(Vector3f position) {
         this.position = position;
     }
 
+    /**
+     * Sending any updated flags (sprinting, onFire, etc.) to the client while in spectator is not needed
+     * Also "fixes" https://github.com/GeyserMC/Geyser/issues/3318
+     */
     @Override
     public void setFlags(ByteEntityMetadata entityMetadata) {
-        super.setFlags(entityMetadata);
-        session.setSwimmingInWater((entityMetadata.getPrimitiveValue() & 0x10) == 0x10 && getFlag(EntityFlag.SPRINTING));
+        // TODO: proper fix, BDS somehow does it? https://paste.gg/p/anonymous/3adfb7612f1540be80fa03a2281f93dc (BDS 1.20.13)
+        if (!this.session.getGameMode().equals(GameMode.SPECTATOR)) {
+            super.setFlags(entityMetadata);
+            session.setSwimmingInWater((entityMetadata.getPrimitiveValue() & 0x10) == 0x10 && getFlag(EntityFlag.SPRINTING));
```

Comment:
Can `0x10` be a constant or is there precedent elsewhere for a constant not being used (in which case its fine)?

Reply:
I'll be honest, no idea
that's the part of the existing code I left untouched; just made sure it's not sent in spectator mode since any flag updates while in it seem to cause the chunk reloading issue - plus, those flags (swimming, on fire, sprinting, etc) are not needed while in spectator mode

Reply:
Ah whoops, I overlooked reading the red. all good then.